security/engine/engine.go

```diff
@@ -0,0 +1,150 @@
+/*
+ * Copyright 2020 gRPC authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// Package engine contains functions/structs for CEL engine.
+package engine
+
+import (
+	"github.com/envoyproxy/go-control-plane/envoy/config/rbac/v2"
+	"google.golang.org/genproto/googleapis/api/expr/v1alpha1"
+	"google.golang.org/grpc/peer"
+)
+
+// Decision enum
+type Decision int32
+
+const (
+	DECISION_ALLOW		Decision = 0
+	DECISION_DENY  		Decision = 1
+	DECISION_UNKNOWN	Decision = 2
+)
+
+var Decision_name = map[int32]string{
+	0: "ALLOW",
+	1: "DENY",
+	2: "UNKNOWN",
+}
+
+var Decision_value = map[string]int32{
+	"ALLOW": 	0,
+	"DENY":  	1,
+	"UNKNOWN": 	2,
+}
+
+// output of CEL engine
+type AuthorizationDecision struct {
+	decision 				Decision
+	authorizationContext	string
+}
+
+// Action enum
+type Action int32
+
+const (
+	ACTION_ALLOW	Action = 0
+	ACTION_DENY  	Action = 1
+)
+
+var Action_name = map[int32]string{
+	0: "ALLOW",
+	1: "DENY",
+}
+
+var Action_value = map[string]int32{
+	"ALLOW": 	0,
+	"DENY":  	1,
+}
+
+// struct for CEL expressions
+type Condition struct {
+	expr	*expr.Expr
+}
+
+func (condition Condition) matches(peerInfo *peer.Peer) bool {
```

Comment:
We also need Metadata. Consider combine these two fields into a struct and make it something like "AuthorizationArgs"